src/main/java/reposense/commits/CommitInfoAnalyzer.java

```diff
@@ -93,9 +101,65 @@ public static CommitResult analyzeCommit(CommitInfo commitInfo, RepoConfiguratio
             extractTagNames(tags);
         }
 
-        int insertion = getInsertion(statLine);
-        int deletion = getDeletion(statLine);
-        return new CommitResult(author, hash, date, messageTitle, messageBody, tags, insertion, deletion);
+        if (statLine.isEmpty()) { // empty commit, no files changed
+            return new CommitResult(author, hash, date, messageTitle, messageBody, tags, 0, 0, new HashMap<>());
+        }
+
+        String[] statInfos = statLine.split(NEW_LINE_SPLITTER);
+        String[] fileTypeContributions = Arrays.copyOfRange(statInfos, 0, statInfos.length - 1);
+        Map<FileType, Map<String, Integer>> fileTypeAndContributionMap =
+                getFileTypesAndContribution(fileTypeContributions, config);
+
+        String contributionStat = statInfos[statInfos.length - 1]; // last index is the file contribution statistics
+        int totalCommitInsertions = getInsertion(contributionStat);
+        int totalCommitDeletions = getDeletion(contributionStat);
+
+        return new CommitResult(author, hash, date, messageTitle, messageBody, tags, totalCommitInsertions,
+                totalCommitDeletions, fileTypeAndContributionMap);
+    }
+
+    /**
+     * Extract the additions and deletions of file types that have been modified
+     */
+    private static Map<FileType, Map<String, Integer>> getFileTypesAndContribution(String[] filePathContributions,
+            RepoConfiguration config) {
+        Map<FileType, Map<String, Integer>> fileTypesAndContribution = new HashMap<>();
```

Comment:
Since the inner `Map<String, Integer>` only contains two possible keys, perhaps it could be extracted into a new model?

Reply:
Yep extracting to a new model will be much cleaner.

Extracted this to a new model `ContributionPair`. Note that there are other places where we can utilise this `ContributionPair` class such as `AuthorDailyContribution`, `CommitResult` etc. (since they both use insertion and deletion variables). But this can be a future PR to refactor them to use `ContributionPair`.

Reply:
If this is really the case, can create an issue on this. Will help keep a track.

Reply:
Will do when this is merged